### PR TITLE
[scan] add possibility for copying pre-built .xctestrun file into test_output directory

### PR DESCRIPTION
**Pull request by Kukurijek**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context

Resolves #18056
### Description

Please read #18056
If this parameter is set to true, the .xctestrun file will be copied into test_output directory and will be called settings.xctestrun.
### Testing Steps

You need an iOS example project with a scheme for UI tests.
Configure a scan lane with parameter `additional_xctestrun: true`.
Use this fastlane version as described [here](https://github.com/fastlane/fastlane/blob/master/Testing.md#test-your-local-fastlane-code-base-with-your-setup).

**Note:** 
If you get a timeout, increase the timeout and the number of retries with the variables:
ENV["FASTLANE_XCODEBUILD_SETTINGS_TIMEOUT"]
ENV["FASTLANE_XCODE_LIST_TIMEOUT"]
ENV["FASTLANE_XCODEBUILD_SETTINGS_RETRIES"]

If you get a lint error, please remove lint from the project, because [Fastlane.swift](https://github.com/fastlane/fastlane/blob/master/fastlane/swift/Fastlane.swift)  has some lines with more than 120 characters.

I have already tested, but please review :)

### CODE REVIEW

**Comment by Kukurijek** [comment]
@googlebot I signed it!

**Comment by minuscorp** [comment]
I don't think you should push and update the Swift files, as they're auto generated when the fastlane release process is done.

**Comment by Kukurijek** [comment]
> I don't think you should push and update the Swift files, as they're auto generated when the fastlane release process is done.

I removed the changes in .swift files.
Thanks!

**Comment by minuscorp** [comment]
Make sure you run locally the lane `generate_swift_api` to see if the changes are well passed into the Swift project as well as the SPM package 🙏

**Comment by Kukurijek** [comment]
> Make sure you run locally the lane `generate_swift_api` to see if the changes are well passed into the Swift project as well as the SPM package 🙏

Done.
The changes look good!
What do you mean with "SPM package"? (Brewfile.lock.json and Gemfile.lock have no problem)

Thanks a lot for the review!

**Comment by minuscorp** [comment]
There's a step in that lane where the fastlane's SPM package gets build, so if you could run it without issues it's fine!

File: scan/lib/scan/options.rb [hunk 1]
```diff
@@ -269,6 +269,12 @@ def self.available_options
                                      optional: true,
                                      is_string: false,
                                      default_value: false),
+        FastlaneCore::ConfigItem.new(key: :additional_xctestrun,
+                                     env_name: "SCAN_SHOULD_ADDITIONAL_XCTESTRUN",
+                                     description: "Should provide additional copy of .xctestrun file (settings.xctestrun) and place in output path?",
+                                     optional: true,
```
Comment (minuscorp) [inline scan/lib/scan/options.rb:275]: Add `type: Boolean`

**Review by minuscorp — COMMENTED** [review]
This PR should add a test in order to confirm the expected behavior, for the rest, seems fine

File: scan/lib/scan/options.rb (hunk 1 above)
Comment (Kukurijek) [reply scan/lib/scan/options.rb:276]: Done.
Lane generate_swift_api is ok.
I have seen that some other arguments do not have a "type", so I thought that parameter "is_string" is enough.
Could you please retry CI jobs, locally is everything ok.

Comment (minuscorp) [reply scan/lib/scan/options.rb:276]: Try to add a test case that addresses the changes if you can 🙏

**Comment by joshdholtz** [comment]
@googlebot I consent.

**Comment by joshdholtz** [comment]
@Kukurijek Heyyyy 👋  I fixed up the unit tests and also renamed the variable from `additional_xctestrun` to `output_xctestrun`. Hopefully that’s okay 😇  Do you mind taking a look at this before it gets merged? 🙏

**Comment by Kukurijek** [comment]
@joshdholtz Hi!  👋  👋 That is really kind of you 🥇 
Sorry for keeping the PR open so long, due to lack of time and zero knowledge of Ruby I did not get to finish it 😅
That is great, thank you really much 👍 
Yes, of course, I'll test it and get back to you!

**Comment by Kukurijek** [comment]
@joshdholtz It looks good! I updated just the name of parameter in the unit test description and merged master.
Thanks again 👍

**Review by minuscorp — APPROVED** [review]
Perfect! LGTM! 🚀

**Review by joshdholtz — APPROVED** [review]
Woot! Thanks for doing all the work here and testing my small change ❤️ Really appreciate the contribution here

**Pull request merged**